app/src/main/java/com/dgaf/happyhour/Model/RestaurantModel.java

```diff
@@ -53,7 +72,23 @@ public String getWebsite() {
         return getString("website");
     }
 
+    public double getDistanceFrom(ParseGeoPoint location) {
+        ParseGeoPoint restaurant = getParseGeoPoint("location");
+
+        if (restaurant == null) {
+            return 0.0;
+        }
+        return location.distanceInMilesTo(restaurant);
+    }
+
     public AvailabilityModel getAvailability() {
+        JSONObject avail = getJSONObject("availability");
+
+        if (avail == null) {
+            return new AvailabilityModel();
+        }
+
```

Comment:
Should add a comment about how this hack works since it gave us such a headache.  The getJSONObject returns null the first time, but we have to return an AvailabilityModel object, so we just return an empty one.  The second time getJSONObject actually returns the correct JSONObject so we create the AvailabilityModel accordingly.